### PR TITLE
Loginpage

### PR DESCRIPTION
**Pull request by YichenLi44**
implement functions of login in and send token to backend and create token. store the username and user's id to datebase

### CODE REVIEW

File: code/server/ai/ai.js
Comment (nichnarmada) [inline code/server/ai/ai.js:1]: why do you need these changes here? just wondering

File: code/server/controller/login.js [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
-
-module.exports.welcome = (req, res) => {
-  const { user } = req.body
-  if (user.verified) {
-    res.json({ msg: 'You have logged in!' })
+const jwt = require('jsonwebtoken');
+const User = require('../model/users.js');
+// Import your User model
+
+const secretKey = 'testestest';
```
Comment (nichnarmada) [inline code/server/controller/login.js:5]: is this the real secret key?

File: code/server/middleware/requireLogin.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
-import jwt from 'jsonwebtoken'
+const jwt = require('jsonwebtoken');
+const secretKey = 'testestest';
```
Comment (nichnarmada) [inline code/server/middleware/requireLogin.js:2]: is this also the real secret key?

File: code/server/package.json
Comment (nichnarmada) [inline code/server/package.json:1]: why remove the type module? what happens if you don't remove it?

**Review by nichnarmada — CHANGES_REQUESTED** [review]
Check my comments pls

File: code/server/ai/ai.js
Comment (guanchengBU) [reply code/server/ai/ai.js:1]: I forgot to change it back. It had some module error at first, so I change everything to "require".

File: code/server/controller/login.js (hunk 1 above)
Comment (guanchengBU) [reply code/server/controller/login.js:5]: secret key can be anything. it's used to generate a access token. I can change the key if needed

File: code/server/ai/ai.js
Comment (nichnarmada) [reply code/server/ai/ai.js:1]: This one is resolved, so dw

**Review by nichnarmada — APPROVED** [review]
looks good~

**Pull request merged**